pydantic/env_settings.py

```diff
@@ -1,12 +1,16 @@
 import os
 import warnings
-from typing import Any, Dict, Iterable, Mapping, Optional
+from pathlib import Path
+from typing import Any, Dict, Iterable, Optional, Union
 
 from .fields import ModelField
 from .main import BaseModel, Extra
+from .read_env_file import read_env_file
 from .typing import display_as_type
 from .utils import deep_update
 
+env_file_sentinel: str = object()  # type: ignore
```

Comment:
on second thoughts I think this is a big weird/dangerous. Best to use the ellipsis and add `EllipsisType` to the union as described above.

Reply:
Tried that, didn't work. mypy complains about it not being a type when I try `EllipsisType = type(...)` or `type(Ellipsis)`.

Reply:
odd, okay then probably best to use `env_file_sentinel = str(object())`, looks weird, but I think should work well.

Or you can just type ignore the function definition.